packages/contracts/test/ClaimReportPosRep.test.ts

```diff
@@ -0,0 +1,237 @@
+    flattenProof,
+    genNullifier,
+    genProofAndVerify,
+    genReportNullifierCircuitInput,
+    genUserState,
+    genVHelperIdentifier,
+} from './utils'
+
+describe('Claim Report Positive Reputation Test', function () {
+    this.timeout(1000000)
+    let unirep: Unirep
+    let app: UnirepApp
+    let chainId: number
+    let user: IdentityObject
+
+    let snapshot: any
+    const epochLength = 300
+    const posReputation = 3
+    const circuit = 'reportNullifierProof'
+    const identifier = genVHelperIdentifier(
+        'reportNullifierProofVerifierHelper'
+    )
+    let usedPublicSig: any
+    let usedProof: any
+
+    {
+        before(async function () {
+            snapshot = await ethers.provider.send('evm_snapshot', [])
+        })
+        after(async function () {
+            await ethers.provider.send('evm_revert', [snapshot])
+        })
+    }
+
+    before(async function () {
+        try {
+            const [deployer] = await ethers.getSigners()
+            const contracts = await deployApp(deployer, epochLength)
+            unirep = contracts.unirep
+            app = contracts.app
+
+            user = createRandomUserIdentity()
+
+            chainId = await unirep.chainid()
+
+            const userState = await genUserState(user.id, app)
+            const { publicSignals, proof } =
+                await userState.genUserSignUpProof()
+            await app.userSignUp(publicSignals, proof, user.hashUserId, false)
+            await app
+                .userRegistry(user.hashUserId)
+                .then((res) => expect(res).to.be.true)
+            console.log('user register success...')
+            userState.stop()
+        } catch (err) {
+            console.error(err)
+        }
+    })
+
+    it('should succeed with valid inputs (correct epochKey, epoch, etc)', async () => {
+        const reportId = 0
+        const currentNonce = 0
+        const hashUserId = user.hashUserId
+
+        const attesterId = BigInt(app.address)
+        const userState = await genUserState(user.id, app)
+        const currentEpoch = await userState.sync.loadCurrentEpoch()
+
+        const reportNullifier = genNullifier(hashUserId, reportId)
+        const reportNullifierCircuitInputs = genReportNullifierCircuitInput({
+            reportNullifier,
+            hashUserId,
+            reportId,
+            currentEpoch,
+            currentNonce,
+            attesterId,
+            chainId,
+        })
+
+        const { publicSignals, proof } = await genProofAndVerify(
+            circuit,
+            reportNullifierCircuitInputs
+        )
+
+        usedPublicSig = publicSignals
+        usedProof = flattenProof(proof)
+
+        const tx = await app.claimReportPosRep(
+            usedPublicSig,
+            usedProof,
+            identifier,
+            posReputation
+        )
+        await expect(tx)
+            .to.emit(app, 'ClaimPosRep')
+            .withArgs(publicSignals[0], currentEpoch)
```

Comment:
Maybe we can find a way to access the reputation in userState and check the value of reputation?